examples/Sinusoid/sinusoid_example.jl

```diff
@@ -0,0 +1,100 @@
+# # Fitting parameters of a sinusoid
+
+# First, we load the packages we need:
+using LinearAlgebra, Random
+
+using Distributions, Plots
+
+using EnsembleKalmanProcesses
+using EnsembleKalmanProcesses.ParameterDistributions
+const EKP = EnsembleKalmanProcesses
+nothing # hide
+
+# Now, we define a model which generates a sinusoid given parameters ``theta``: an
+# amplitude and a vertical shift. We will estimate these parameters from data.
+# Each curve is generated with a random phase shift.
+dt = 0.01
+trange = 0:dt:(2 * pi + dt)
+function model(amplitude, vert_shift)
+    phi = 2 * pi * rand()
+    return amplitude * sin.(trange .+ phi) .+ vert_shift
+end
+nothing # hide
+
+# Seed for pseudo-random number generator.
+rng_seed = 41
+rng = Random.MersenneTwister(rng_seed)
+nothing # hide
+
+# We then define ``G(\theta)``, which returns the observables of the sinusoid
+# given a parameter vector. These observables should be defined such that they
+# are informative about the parameters we wish to estimate. Here, the two
+# observables are the ``y`` range of the curve (which is informative about its
+# amplitude), as well as its mean (which is informative about its vertical shift).
+function G(u)
+    theta, vert_shift = u
+    sincurve = model(theta, vert_shift)
+    return [maximum(sincurve) - minimum(sincurve), mean(sincurve)]
+end
+nothing # hide
+
+# We now define prior distributions on the two parameters. For the amplitude,
+# we define a Gaussian prior with mean 5 and standard deviation 1. It is
+# additionally constrained to be nonnegative. For the vertical shift we define
+# an unconstrained Gaussian prior with mean 0 and standard deviation 5.
+prior_u1 = constrained_gaussian("amplitude", 5, 1, 0, Inf)
+prior_u2 = constrained_gaussian("vert_shift", 0, 5, -Inf, Inf)
+prior = combine_distributions([prior_u1, prior_u2])
+nothing # hide
+
+# Suppose we have a noisy observation of the true system. Here, we create a
+# pseudo-observation ``y`` by running our model with the correct parameters
+# and adding Gaussian noise to the output.
+dim_output = 2
+
+Γ = 0.1 * I
+noise_dist = MvNormal(zeros(dim_output), Γ)
+
+theta_true = [1.0, 7.0]
+y = G(theta_true) .+ rand(noise_dist)
+nothing # hide
+
+# We now generate the initial ensemble and set up the ensemble Kalman inversion.
+N_ensemble = 5
+N_iterations = 5
+
+initial_ensemble = EKP.construct_initial_ensemble(rng, prior, N_ensemble)
+
+ensemble_kalman_process = EKP.EnsembleKalmanProcess(initial_ensemble, y, Γ, Inversion(); rng = rng)
+nothing # hide
+
+# We are now ready to carry out the inversion. At each iteration, we get the
+# ensemble from the last iteration, apply ``G(\theta)`` to each ensemble member,
+# and apply the Kalman update to the ensemble.
+for i in 1:N_iterations
+    params_i = get_u_final(ensemble_kalman_process)
```

Comment:
This is actually a typo, it must be `get_\phi_final(priors, ensemble_kalman_process)` (as you have used a lognormal prior in the prior, so get_u_final will return the unconstrained parameters).

Reply:
I guess it still works because of the problem spec having high mass far from the boundary